Lesson4.2/src/com/home/model/MilitaryOffice.java

```diff
@@ -0,0 +1,84 @@
+package com.home.model;
+
+import com.home.service.PersonsRegistry;
+
+import java.util.LinkedList;
+import java.util.List;
+
+public class MilitaryOffice {
+    private final PersonsRegistry registry;
+    private final List<MilitaryUnit> militaryUnit;
+
+
+    public MilitaryOffice(PersonsRegistry registry, List<MilitaryUnit> militaryUnit) {
+        this.registry = registry;
+        this.militaryUnit = militaryUnit;
+    }
+
+    public int getNumberOfRequiredRecruits() {
+        int requiredNumberOfRecruits = 0;
+        for (MilitaryUnit unit : militaryUnit) {
+            requiredNumberOfRecruits += unit.getNumberOfFreePlaces();
+        }
+        return requiredNumberOfRecruits;
+    }
+
+    public List<Person> distributeRecruits(Address address) {
+        List<Person> listOfMen = new LinkedList<>();
+        for (Person person : registry.getListOfPeople(address)) {
+            if (checkForSuitability(person)) {
+                listOfMen.add(person);
+            }
+        }
+        return listOfMen;
+    }
+
+    public void distributeRecruits(String country) {
+        List<Person> listOfMen = new LinkedList<>();
+        for (Person person : registry.getListOfPeople(country)) {
+            if (checkForSuitability(person)) {
+                listOfMen.add(person);
+            }
+        }
+        for (Person recruit : listOfMen) {
+            int freePlaces = 0;
+            for (MilitaryUnit unit : militaryUnit) {        // distribution in units one by one
+                if (unit.getNumberOfFreePlaces() > 0) {
+                    unit.addNewRecruit(recruit);
+                    break;
+                }
+            }
+            for (MilitaryUnit unit : militaryUnit) {
+                freePlaces = unit.getNumberOfFreePlaces();
+            }
+            if (freePlaces == 0) {
```

Comment:
this is not correct. if (freePlaces == 0)  that would mean only that the very last element of militaryUnit list has no free places. You might want to sum all free places in the for loop above to make sure that there are no places at all